centreon/src/Core/Dashboard/Application/UseCase/FindMetricsTop/FindMetricsTopRequest.php

```diff
@@ -23,7 +23,17 @@
 
 namespace Core\Dashboard\Application\UseCase\FindMetricsTop;
 
+use Symfony\Component\Validator\Constraints as Assert;
+
 final class FindMetricsTopRequest
 {
-    public string $metricName = '';
+    /**
+     * @param string $metricName
+     */
+    public function __construct(
+        #[Assert\NotBlank]
+        #[Assert\Type('string')]
+        public readonly string $metricName
```

Comment:
Useless assertion OR wrong typing

Depends whether or not it is possible to send a payload with anything else than a string in $metricName

Reply:
Done